projects/storefrontlib/cms-components/product/product-intro/product-intro.component.ts

```diff
@@ -63,6 +57,47 @@ export class ProductIntroComponent implements AfterContentChecked {
     return this.winRef.document.querySelector('cx-product-reviews');
   }
 
+  /**
+   * Create observable that check reviews status.
+   * Emits true if SSR or reviews component is defined.
+   * Else, observe whether reviews are visible after mutation.
+   */
+  protected observeReviews(): Observable<boolean> {
+    return new Observable((subscriber) => {
+      if (!!this.getReviewsComponent() || !this.winRef.isBrowser()) {
+        subscriber.next(true);
+      } else {
+        const observer = this.observePageLayoutMutation(subscriber);
+        // disconnect observer on unsubscribe
+        return () => {
+          observer.disconnect();
+        };
+      }
+    });
```

Comment:
According to [line 91](https://github.com/SAP/spartacus/pull/15438/files#diff-e97f441f4a53211ec46be20345d362788a4570c07e291ecbdea671021bb464dfR91), a new emission will be passed to subscriber every time when the MutationObserver callback is invoked.

Maybe we should add here `pipe(distinctUntilChanged)` here, so to avoid unnecessary emissions when the apperance of reviewscomponent doesn't change (in other words: when it's still `true` or still `false`), so to avoid unnecessary emissions triggering `async` pipe and rerenders.

Reply:
Good point! Anyway, problem is outdated since we changed our approach after discussion.